Assets/Scripts/Game/LevitateMotor.cs

```diff
@@ -68,17 +68,25 @@ private void Update()
             if (GameManager.Instance.PlayerEntity.IsParalyzed)
                 return;
 
+            float inputX = 0f;
+            float inputY = 0f;
             // Forward/backwards
             if (InputManager.Instance.HasAction(InputManager.Actions.MoveForwards))
-                AddMovement(playerCamera.transform.forward);
-            else if (InputManager.Instance.HasAction(InputManager.Actions.MoveBackwards))
-                AddMovement(-playerCamera.transform.forward);
+                inputY++;
+            if (InputManager.Instance.HasAction(InputManager.Actions.MoveBackwards))
+                inputY--;
 
             // Right/left
             if (InputManager.Instance.HasAction(InputManager.Actions.MoveRight))
-                AddMovement(playerCamera.transform.right);
-            else if (InputManager.Instance.HasAction(InputManager.Actions.MoveLeft))
-                AddMovement(-playerCamera.transform.right);
+                inputX++;
+            if (InputManager.Instance.HasAction(InputManager.Actions.MoveLeft))
+                inputX--;
+
+            if (inputX != 0.0f || inputY != 0.0f)
+            {
+                float inputModifyFactor = (inputX != 0.0f && inputY != 0.0f && playerMotor.limitDiagonalSpeed) ? .7071f : 1.0f;
```

Comment:
Consider putting both components in a 2d vector and normalizing it. If we ever want to add joystick support

Reply:
Normalisation is what I used in the first commit, then I aligned implementation with the other "motor" classes.
I know nothing about controllers support, that's why I asked feedback from jfetienne.